### PR TITLE
chore: try to bump jemalloc and fix musl link

### PR DESCRIPTION
**Pull request by PsiACE**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

- bump jemalloc
- `RUSTFLAGS="-C target-feature=-crt-static"`, thanks @Xuanwo
- rm '--compress-debug-sections=zlib-gabi', it is not a recognized feature for this target (ignoring feature)

## Changelog

- Not for changelog (changelog entry is not required)

## Related Issues

Fixes #issue

### CODE REVIEW

File: .github/workflows/databend-release.yml [hunk 1]
```diff
@@ -94,7 +94,7 @@ jobs:
       - name: Set musl rustflags
         if: matrix.platform == 'musl'
         run: |
-          flags="-C link-arg=-Wl,--compress-debug-sections=zlib-gabi"
```
Comment (everpcpc) [inline .github/workflows/databend-release.yml:97]: `--compress-debug-sections=zlib-gabi` is still needed?

File: .github/actions/build_linux/action.yml [hunk 2]
```diff
@@ -16,6 +16,13 @@ runs:
         registry: public.ecr.aws/i7g1w5q7
         image: ${{ inputs.target }}
 
+    - name: Set musl rustflags
+      if: inputs.target == 'x86_64-unknown-linux-musl' || inputs.target == 'aarch64-unknown-linux-musl'
```
Comment (everpcpc) [inline .github/actions/build_linux/action.yml:20]:
```suggestion
      if: contains(inputs.target, 'musl')
```

File: .github/workflows/databend-release.yml [hunk 3]
```diff
@@ -94,7 +94,7 @@ jobs:
...
+          flags="-C link-arg=-W -C target-feature=-crt-static"
```
Comment (PsiACE) [inline .github/workflows/databend-release.yml:97]: I need a little time to verify this

File: .github/workflows/databend-release.yml (hunk 1 above)
Comment (PsiACE) [reply .github/workflows/databend-release.yml:97]: I think it won't affect the build. Remove it temporarily, x86_64 does not seem to recognise it correctly.

File: .github/workflows/databend-release.yml (hunk 3 above)
Comment (PsiACE) [reply .github/workflows/databend-release.yml:97]: It compiles successfully on my laptop with build-tool musl images, @everpcpc  can you help to confirm?

Comment (Xuanwo) [reply .github/workflows/databend-release.yml:97]: Can this binary be executed successfully? I'm failed to build on my PC for:

```shell
  = note: /usr/bin/ld: /home/xuanwo/Code/datafuselabs/databend/target/x86_64-unknown-linux-musl/debug/deps/libminstant-5b533bdc94413bb4.rlib(minstant-5b533bdc94413bb4.minstant.713ea42f-cgu.13.rcgu.o): in function `std::f64::<impl f64>::ceil':
          /rustc/cd282d7f75da9080fda0f1740a729516e7fbec68/library/std/src/f64.rs:68: undefined reference to `ceil'
          /usr/bin/ld: /home/xuanwo/Code/datafuselabs/databend/target/x86_64-unknown-linux-musl/debug/deps/libminstant-5b533bdc94413bb4.rlib(minstant-5b533bdc94413bb4.minstant.713ea42f-cgu.13.rcgu.o): in function `std::f64::<impl f64>::round':
          /rustc/cd282d7f75da9080fda0f1740a729516e7fbec68/library/std/src/f64.rs:88: undefined reference to `round'
          collect2: error: ld returned 1 exit status
```

But I think it's not related to this PR, instead, it's resulted from https://github.com/rust-lang/rust/issues/61328

It's better to run our test suits on musl binaries too (maybe in a new PR).

Comment (everpcpc) [reply .github/workflows/databend-release.yml]: @Xuanwo try with build-tool image?
`IMAGE='datafuselabs/build-tool:x86_64-unknown-linux-musl' RUSTFLAGS='-C link-arg=-Wl,--compress-debug-sections=zlib-gabi -C target-feature=-crt-static' ./scripts/setup/run_build_tool.sh cargo build --target x86_6
4-unknown-linux-musl`

**Review by Xuanwo — APPROVED** [review]

**Review by everpcpc — APPROVED** [review]

**Pull request merged**